pkg/endpoint/types.go

```diff
@@ -14,19 +14,19 @@ import (
 
 // Provider is an interface to be implemented by components abstracting Kubernetes, and other compute/cluster providers
 type Provider interface {
-	// Retrieve the IP addresses comprising the given service.
+	// ListEndpointsForService retrieves the IP addresses comprising the given service.
 	ListEndpointsForService(service.MeshService) []Endpoint
 
 	// ListEndpointsForIdentity retrieves the list of IP addresses for the given service account
-	ListEndpointsForIdentity(identity.K8sServiceAccount) []Endpoint
+	ListEndpointsForIdentity(serviceIdentity identity.ServiceIdentity) []Endpoint
```

Comment:
Since this is an interface definition, I believe this should be 
`ListEndpointsForIdentity(identity.ServiceIdentity) []Endpoint`

Reply:
Corrected!